data/result-exporter/swe-cne-exporter/src/main/java/com/farao_community/farao/data/swe_cne_exporter/SweRemedialActionSeriesCreator.java

```diff
@@ -128,30 +130,34 @@ public List<RemedialActionSeries> generateRaSeriesReference(Contingency continge
 
     private RemedialActionSeries generateRaSeries(State state, RemedialActionSeriesCreationContext context, boolean onlyReference) {
         RaoResult raoResult = sweCneHelper.getRaoResult();
-        AngleMonitoringResult angleMonitoringResult = sweCneHelper.getAngleMonitoringResult();
         Crac crac = sweCneHelper.getCrac();
         List<RemedialAction<?>> usedRas = new ArrayList<>();
-        if (Objects.nonNull(raoResult)) {
-            context.getCreatedIds().stream().sorted()
-                    .map(crac::getRemedialAction)
-                    .filter(ra -> raoResult.isActivatedDuringState(state, ra))
-                            .forEach(usedRas::add);
+        if (Objects.isNull(raoResult)) {
+            return null;
         }
-        if (Objects.nonNull(angleMonitoringResult) && !angleMonitoringResult.isDivergent()) {
+        context.getCreatedIds().stream().sorted()
+                .map(crac::getRemedialAction)
+                .filter(ra -> {
+                    if (raoResult instanceof RaoResultImpl) {
+                        return raoResult.isActivatedDuringState(state, ra);
+                    } else if (raoResult instanceof RaoResultWithAngleMonitoring) {
+                        return ((RaoResultWithAngleMonitoring) raoResult).isActivatedDuringStateButNotByAngleMonitoring(state, ra);
+                    }
```

Comment:
Again, the SWE CNE exporter must not know that the RaoResultWithAngleMonitoring exists.
The "isActivatedDuringState" in RaoResultWithAngleMonitoring now returns RAs activated by RAO or by angle monitoring. So you don't need to separate things anymore (first add RAs from RaoResult, then add them from AngleMonitoring), the code should become simpler than before (only use RAs in RaoReult).